tracing-subscriber/src/layer.rs

```diff
@@ -773,42 +787,63 @@ impl<'a, S: Subscriber> Context<'a, S> {
             .unwrap_or(false)
     }
 
-    /// Visits every span in the current context with a closure.
+    /// Returns [stored data] for the span that the wrapped subscriber considers
+    /// to be the current.
+    ///
+    /// If this returns `None`, then we are not currently within a span.
+    ///
+    /// **Note**: This requires the wrapped subscriber to implement the
+    /// [`LookupSpan`] trait. `Layer` implementations that wish to use this
+    /// function can bound their `Subscriber` type parameter with
+    /// ```rust,ignore
+    /// where S: Subscriber + for<'span> LookupSpan<'span>,
+    /// ```
+    /// or similar.
     ///
-    /// The provided closure will be called first with the current span,
-    /// and then with that span's parent, and then that span's parent,
-    /// and so on until a root span is reached.
-    pub fn scope<E, F>(&self, mut f: F) -> Result<(), E>
+    /// [stored data]: ../registry/struct.SpanRef.html
+    /// [`LookupSpan`]: ../registry/trait.LookupSpan.html
+    #[inline]
+    #[cfg(feature = "registry")]
+    pub fn lookup_current(&self) -> Option<registry::SpanRef<'_, S>>
```

Comment:
on the fence about this name; if anyone can think of anything better i'd be happy to consider renaming it.

Reply:
I think this one is pretty good. Here's an alternative that I'm not sold on: call it `lookup_current_span` to match with [Context::current_span](https://docs.rs/tracing-subscriber/0.2.0-alpha.1/tracing_subscriber/layer/struct.Context.html#method.current_span). While their return types are different, the `lookup` modifier indicates a `SpanRef<'_, S>` is being returned, not a `span::Current`.

Reply:
I also considered `current_span_data` or `current_data`, if you think those are clearer?

Reply:
I don't think those are clearer because:

1. `current_span_data` loses the implication that its connected to `LookupSpan`.
2. I have the same concern with `current_data`. Additionally, _what_ data the method is looking up is not clear from the name.

Of those options, I like `lookup_current` or `lookup_current_span` the most.

Reply:
I'm happy to change it to `lookup_current_span`; it just feels a little bit wordy.

The one thing is that we currently have a function `Context::span`
https://github.com/tokio-rs/tracing/blob/b2968c8d34bc743ed7da8a0193b41020adf4a983/tracing-subscriber/src/layer.rs#L760-L762
which also returns a `SpanRef`, so it feels a _little_ inconsistent for this one to have `lookup` and that one to not have `lookup`. We _could_ rename it to `lookup_span`, but I like how succinct just `span` is...

Reply:
Oh, I knew I forgot about something. `lookup_current` is best then.